docs/05-storage-providers.md

```diff
@@ -166,225 +174,241 @@ Amazon charges slightly more for non-US buckets. Known bucket locations:
+
+URL Format:
 
 `amzcd://folder/subfolder`
 
-Supported options:
+Options:
 
-* `--authid (Password)`  
-The authorization code.  
+* `--authid (Password)`
+The authorization code.
 The authorization token retrieved from https://duplicati-oauth-handler.appspot.com?type=amzcd
-* `--amzcd-labels (String)`  
-The labels to set.  
-Use this option to set labels on the files and folders created  
+* `--amzcd-labels (String)`
+The labels to set.
+Use this option to set labels on the files and folders created
 Default value: `duplicati,backup`
-* `--amzcd-consistency-delay (Timespan)`  
-The consistency delay.  
-Amazon Cloud drive needs a small delay for results to stay consistent.  
+* `--amzcd-consistency-delay (Timespan)`
+The consistency delay.
+Amazon Cloud drive needs a small delay for results to stay consistent.
 Default value: `15s`
 
 ## Azure blob
 
-This backend can read and write data to Azure blob storage. Allowed format:
+This backend can read and write data to Azure blob storage.  To configure this backend you must create a storage account in the [Azure Portal](https://portal.azure.com) and get an access key by going to `Settings > Access keys` on the storage account details page.  The configuration options can either be provided as part of the URL or as separate parameters.  If both are provided, the URL parameters will take precedence.
 
-`azure://bucketname`
+URL Format:
 
-Supported options:
+`azure://containerName`
+`azure://storageAccountName@containerName`
+`azure://storageAccountName:accessKey@containerName`
 
-* `--azure_account_name (String)`  
-The storage account name.  
-The Azure storage account name which can be obtained by clicking the "Manage Access Keys" button on the storage account dashboard.
-* `--azure_access_key (Password)`  
-The access key.  
-The Azure access key which can be obtained by clicking the "Manage Access Keys" button on the storage account dashboard.
-* `--azure_blob_container_name (String)`  
-The name of the storage container.  
-All files will be written to the container specified.
-* `--auth-password (Password)`  
-Supplies the password used to connect to the server.  
-The password used to connect to the server. This may also be supplied as the environment variable `AUTH_PASSWORD`.
-* `--auth-username (String)`  
-Supplies the username used to connect to the server.  
-The username used to connect to the server. This may also be supplied as the environment variable `AUTH_USERNAME`.
+Options:
+
+* `--azure_account_name (String)`
+The storage account name.
+The name of the Azure storage account to connect to.  This parameter will be ignored if the name is specified in the URL.
+* `--azure_access_key (Password)`
+The storage account access key.
+The access key used to authenticate calls to the Azure storage account.  This parameter will be ignored in the access key is specified in the URL.
+* `--auth-username (String)`
+The storage account name.

```

Comment:
Given how these comments are written, it seems as though the intent is that they're auto-generated from the source, but it wasn't obvious to me if that was a one-time thing, or something that happens regularly.  If it's auto-generated, I'm happy to update the source using these same changes.

These two parameters (`--auth-password` and `--auth-username`) are an interesting story though.  The current documentation is really not helpful to users.  You have to just know that they map to the storageAccount (as opposed to the container name) and the accessKey by looking at the code.  So, it might be reasonable to just remove these entirely.  However, this IS the only method of supplying those parameters using environment variables, so I didn't want to remove it.  But maybe a more generic way of supplying parameters with environment variables in the right change to make.